includes/controls/repeater.php

```diff
@@ -181,4 +182,20 @@ public function content_template() {
 		<# } #>
 		<?php
 	}
+
+	public function validate( array $control_data ): bool {
+		if ( isset( $control_data['min_items'] ) ) {
+
+			if (
+				! isset( $control_data['default'] ) ||
+				count( $control_data['default'] ) < $control_data['min_items']
+			) {
+				throw new \Exception(
+					__( 'In a Repeater control, if you specify a minimum number of items, you must also specify a default value that contains at least that number of items.', 'elementor' )
+				);
+			}
+		}
+
+		return true;
+	}
```

Comment:
@nicoladj77 I am wondering if we should do this validation client side?

See: `elementor/assets/dev/js/editor/components/validator/number.js` as an example. This might not work specific for our scenario, but maybe just follow the same logic.

I am just wondering if it is dangerous to throw this kind of Exception that would totally break the Editor if a 3rd party does something wrong.

Reply:
@louiswol94 this exception should only be thrown in the context of the Controls_Manager

		if ( $control_type_instance instanceof Has_Validation ) {
			try {
				$control_type_instance->validate( $control_data );
			} catch ( \Exception $e ) {
				_doing_it_wrong( sprintf( '%1$s::%2$s', __CLASS__, __FUNCTION__ ), esc_html( $e->getMessage() ), '3.23.0' );
				return false;
			}
		}

I totally understand your concern, I can of course to this differently, returning an array from validate() with two keys, valid and message. I wanted to run this by you because like this is cleaner from a coding perspective, but if you think it's not safe, let's do it the other way.

Reply:
@louiswol94 regarding FE vs BE, this to me looks similar to

		$control_type_instance = $this->get_control( $control_data['type'] );

		if ( ! $control_type_instance ) {
			_doing_it_wrong( sprintf( '%1$s::%2$s', __CLASS__, __FUNCTION__ ), sprintf( 'Control type "%s" not found.', esc_html( $control_data['type'] ) ), '1.0.0' );
			return false;
		}

which means that if a control is in a not valid state, it should not be added. But let me know if you think differently. if I had to do the check in the FE, I would add rows until they are reach the min_items value.